src/plugins/intel_gna/src/gna_graph_tools.hpp

```diff
@@ -131,18 +135,23 @@ inline typename ExtractRawPtr<T>::raw_ptr_type raw_ptr(T obj) {
 template <class Layer>
 inline InferenceEngine::CNNLayerPtr CNNNetPrevLayerSkipCertain(Layer layer,
                                                                int idx,
-                                                               const std::function<bool(CNNLayerPtr)>& shouldSkip) {
+                                                               const std::function<bool(CNNLayerPtr)>& shouldSkip,
+                                                               bool checkOnly = false) {
     IE_ASSERT(layer != nullptr);
     if (!CNNNetHasPrevLayer(raw_ptr(layer), idx)) {
-        THROW_GNA_EXCEPTION << "Can't find PrevLayer. All layers are skipped.";
+        if (!checkOnly) {
```

Comment:
Maybe it is better to check result of `CNNNetHasPrevLayer` instead of adding one more parameter?

Reply:
As above - reverted.